src/PolicyTool/Program.cs

```diff
@@ -0,0 +1,307 @@
+                       name: "--app-name",
+                       description: "Taget Application name.");
+            appName.IsRequired = true;
+
+            var clientSecret = new Option<string>(
+                       name: "--secret",
+                       description: "Client secret.",
+                       getDefaultValue: () => "ChangeIt");
+            clientSecret.IsRequired = true;
+
+            var rootCommand = new RootCommand("Policy tool");
+            rootCommand.AddGlobalOption(appName);
+
+            var addApp = new Command("add-app", "Add a new application.");
+            addApp.AddOption(clientSecret);
+
+            var showApp = new Command("show-app", "Display app details.");
+
+            var addSecret = new Command("add-secret", "Add a new application secret.");
+            addSecret.AddOption(clientSecret);
+
+            var roles = new Option<string[]>(
+                      name: "--roles",
+                      description: "List of Roles.");
+            roles.IsRequired = true;
+            roles.AllowMultipleArgumentsPerToken = true;
+            var addRoles = new Command("add-roles", "Idempotently add roles to an application");
+            addRoles.AddOption(roles);
+
+            var userName = new Option<string>(
+                      name: "--user-name",
+                      description: "User name.");
+            userName.IsRequired = true;
+            var domain = new Option<string>(
+                      name: "--domain",
+                      description: "Domain.");
+            userName.IsRequired = false;
+            var addUser = new Command("add-user", "Add a new user to the application with the specified roles.");
+            addUser.AddOption(userName);
+            addUser.AddOption(roles);
+            addUser.AddOption(domain);
+
+
+            var mainBus = new Dispatcher("Main Bus");
+            var appSvc = new ReactiveDomain.Policy.ApplicationSvc(EsConnection, mainBus);
+            var appRm = new ReactiveDomain.Policy.ReadModels.ApplicationRm(EsConnection);
+            var polSvc = new PolicySvc(EsConnection, mainBus);
+            var userStore = new UserStore(EsConnection);
+            var validation = new UserValidation(userStore);
+            var userRm = new UsersRm(EsConnection);
+            var subjectsRm = new SubjectsRm(EsConnection);
+            var userSvc = new UserSvc(EsConnection.GetRepository(), mainBus);
+            addApp.SetHandler(
+                (string name, string secret) =>
+                {
+                    var appId = Guid.NewGuid();
+                    var version = "1.0";
+                    var cmd = new ApplicationMsgs.CreateApplication(
+                              appId,
+                              Guid.NewGuid(),
+                              name,
+                              version,
+                              false);
+                    cmd.CorrelationId = Guid.NewGuid();
```

Comment:
Why not use `MessageBuilder` for this, i.e.
```
                    var cmd = MessageBuilder.New(() => new ApplicationMsgs.CreateApplication(
                              appId,
                              Guid.NewGuid(),
                              name,
                              version,
                              false));
```

Reply:
👍